### ISSUE TITLE
[BUG] SOD balance not DB-persisted in live mode — midnight rollover ignored

### ISSUE THREAD

**Issue by vipulbms** [vipulbms/crypto-trader-agent#178]
## What

Session L (#170) fixed the start-of-day (SOD) balance to be DB-persisted in **paper mode** using `_get_or_set_sod_balance()`. This prevents the false kill-switch when `reset_paper.py` runs while the agent is alive, and handles midnight rollovers.

However, the fix is explicitly gated on `mode == "paper" and not is_backtest` (main.py ~line 253). **Live mode still uses the in-memory `start_of_day_bal` captured once at agent startup.**

## Why It Matters

If the live agent runs past midnight without a restart:
- `start_of_day_bal` remains the value from the previous day's startup
- `get_daily_pnl(start_of_day_bal)` computes P&L relative to yesterday's opening balance
- Daily loss limit (10%) can suppress legitimate trades all morning
- Or worse: yesterday's gains mask today's real losses, delaying the 10% kill switch

Example: Agent starts at 09:00 SGT with $1,200. Midnight passes. By 02:00 SGT the next day balance drops to $1,100. Daily loss = ($1,100 − $1,200) / $1,200 = **−8.3%** (correct) — but the agent still uses $1,200 as the baseline, missing the -10% threshold entirely if yesterday was a bigger up day.

## Root Cause

```python
# main.py line ~181 — captured once at startup, never refreshed in live mode
start_of_day_bal = balance_data["total_usd"]

# main.py loop ~line 253 — only paper mode is protected
if mode == "paper" and not is_backtest:
    start_of_day_bal = _get_or_set_sod_balance(paper_db_path, current_bal_for_sod)
```

## How to Fix

1. **`src/storage/database.py`** — ensure `agent_state` table exists in `live_trading.db` (same as paper mode).
2. **`main.py`** — extend the SOD refresh block to cover live mode:
```python
if not is_backtest:
    db_path = paper_db_path if mode == "paper" else live_db_path
    start_of_day_bal = _get_or_set_sod_balance(db_path, current_bal_for_sod)
```
3. **`_get_or_set_sod_balance()`** — confirm it works with `live_trading.db` schema (same `agent_state` table pattern).

## Files to Change
- `main.py` — extend SOD guard to live mode
- `src/storage/database.py` — ensure `agent_state` in live DB schema
- `tests/test_risk_manager.py` or new `tests/test_sod_balance.py` — midnight rollover test for live mode
